builtin/game/falling.lua

```diff
@@ -144,21 +144,35 @@ core.register_entity(":__builtin:falling_node", {
 })
 
 local function convert_to_falling_node(pos, node)
-	local obj = core.add_entity(pos, "__builtin:falling_node")
-	if not obj then
-		return false
-	end
 	node.level = core.get_node_level(pos)
 	local meta = core.get_meta(pos)
 	local metatable = meta and meta:to_table() or {}
+	local oldmeta = metatable.fields
 
 	local def = core.registered_nodes[node.name]
+
+	local ok
+	if def and def.on_fall then
+		ok = def.on_fall(table.copy(pos), table.copy(node))
+		if type(ok) == "boolean" then
+			return ok
+		end
+	end
+
 	if def and def.sounds and def.sounds.fall then
 		core.sound_play(def.sounds.fall, {pos = pos})
 	end
-
+	local obj = core.add_entity(pos, "__builtin:falling_node")
```

Comment:
The entity should be added before playing the sound to ensure there's no sound when it fails.